src/compiler/transformers/es2015.ts

```diff
@@ -162,6 +162,82 @@ namespace ts {
         ReplaceWithReturn,
     }
 
+    type LoopConverter = (node: IterationStatement, outermostLabeledStatement: LabeledStatement, convertedLoopBodyStatements: Statement[]) => Statement;
+
+    // Facts we track as we traverse the tree
+    const enum HierarchyFacts {
+        None = 0,
+
+        // Ancestor facts
+        Function = 1 << 0,                          // Enclosed in a non-arrow function
+        ArrowFunction = 1 << 1,                     // Enclosed in an arrow function
+        AsyncFunctionBody = 1 << 2,                 // Enclosed in an async function body
+        NonStaticClassElement = 1 << 3,             // Enclosed in a non-static, non-async class element
+        CapturesThis = 1 << 4,                      // Enclosed in a function that captures the lexical 'this' (used in substitution)
+        ExportedVariableStatement = 1 << 5,         // Enclosed in an exported variable statement in the current scope
+        TopLevel = 1 << 6,                          // Enclosing block-scoped container is a top-level container
+        Block = 1 << 7,                             // Enclosing block-scoped container is a Block
+        IterationStatement = 1 << 8,                // Enclosed in an IterationStatement
+        IterationStatementBlock = 1 << 9,           // Enclosing Block is enclosed in an IterationStatement
+        ForStatement = 1 << 10,                     // Enclosing block-scoped container is a ForStatement
+        ForInOrForOfStatement = 1 << 11,            // Enclosing block-scoped container is a ForInStatement or ForOfStatement
+        ConstructorWithCapturedSuper = 1 << 12,     // Enclosed in a constructor that captures 'this' for use with 'super'
+        ComputedPropertyName = 1 << 13,             // Enclosed in a computed property name
+        AncestorFactsMask = (ComputedPropertyName << 1) - 1,
+
+        // Subtree facts
+        NewTarget = 1 << 14,                        // Contains a 'new.target' meta-property
+        NewTargetInComputedPropertyName = 1 << 15,  // Contains a 'new.target' meta-property in a computed property name.

```

Comment:
When we compute flags for a member, such as a method or accessor, it both has its own lexical scope (for the parameters and body) as well as shares the lexical scope of its container (due to the computed property name). Without this flag, encountering a NewTarget in a computed property name would incorrectly indicate that NewTarget was set in the body of the method, not in the container.